[JHBuild] Update glib to 2.72.4

dpino opened the pull request:
#### b18feb6f416c7054880a920af213f6dfa864eb7a
<pre>
[JHBuild] Update glib to 2.72.4
<a href="https://bugs.webkit.org/show_bug.cgi?id=257185">https://bugs.webkit.org/show_bug.cgi?id=257185</a>

Reviewed by Michael Catanzaro.

This version of GLib includes a bug fix that was affecting Ubuntu
20.04 containers.

<a href="https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/937">https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/937</a>

* Tools/gtk/jhbuild.modules:
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: <a href="https://commits.webkit.org/264505@main">https://commits.webkit.org/264505@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/58e09d4d7c71474cec54711b7c1f508baf6f86c7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7718 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7989 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8170 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9359 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7875 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7724 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9979 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7910 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10742 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7852 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8968 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7085 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9467 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6278 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7025 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14690 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7434 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7149 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10556 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7641 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6227 "20 flakes 3 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6969 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1874 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11180 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7375 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pgorszkowski-igalia commented on Tools/gtk/jhbuild.modules:
```diff
@@ -222,9 +222,11 @@
          mesonargs="-Dlibmount=disabled -Dselinux=disabled">
-    <branch module="/sources/glib/2.70/glib-${version}.tar.xz" version="2.70.0"
+    <pkg-config>glib-2.0.pc</pkg-config>
+    <branch module="/sources/glib/2.70/glib-${version}.tar.xz"
```
I think module should be 2.71: https://download.gnome.org/sources/glib/2.71/glib-2.71.3.tar.xz

dpino replied:
Good catch! I will amend the patch.

pgorszkowski-igalia commented on Tools/jhbuild/jhbuild-minimal.modules:
```diff
@@ -171,15 +171,13 @@
-    </dependencies>
-    <branch module="/sources/glib/2.70/glib-${version}.tar.xz" version="2.70.0"
+    <pkg-config>glib-2.0.pc</pkg-config>
+    <branch module="/sources/glib/2.70/glib-${version}.tar.xz"
```
I think module should be 2.71: https://download.gnome.org/sources/glib/2.71/glib-2.71.3.tar.xz

pgorszkowski-igalia commented on Tools/wpe/jhbuild.modules:
```diff
@@ -69,12 +69,11 @@
-    </dependencies>
-    <branch module="/sources/glib/2.70/glib-${version}.tar.xz" version="2.70.0"
+    <pkg-config>glib-2.0.pc</pkg-config>
+    <branch module="/sources/glib/2.70/glib-${version}.tar.xz"
```
I think module should be 2.71: https://download.gnome.org/sources/glib/2.71/glib-2.71.3.tar.xz

mcatanzaro reviewed (changes requested):
If you're going to update glib, pick a stable release please (even minor version) rather than a development snapshot (odd minor version). Is there anything wrong with the latest release 2.76.3? Just use that unless you have some reason not to?

dpino committed:
[JHBuild] Update glib to 2.72.4
https://bugs.webkit.org/show_bug.cgi?id=257185

Reviewed by Michael Catanzaro.

This version of GLib includes a bug fix that was affecting Ubuntu
20.04 containers.

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/937

* Tools/gtk/jhbuild.modules:
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: https://commits.webkit.org/264505@main
